spring-cloud-kubernetes-core/src/main/java/org/springframework/cloud/kubernetes/condition/OnKubernetesOrOpenshiftCondition.java

```diff
@@ -0,0 +1,100 @@
+package org.springframework.cloud.kubernetes.condition;
+
+import static org.springframework.cloud.kubernetes.Constants.SPRING_CLOUD_KUBERNETES_CLIENT_PROPS;
+
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.openshift.client.OpenshiftAdapterSupport;
+import java.util.Map;
+import org.springframework.boot.autoconfigure.condition.ConditionMessage;
+import org.springframework.boot.autoconfigure.condition.ConditionOutcome;
+import org.springframework.boot.autoconfigure.condition.SpringBootCondition;
+import org.springframework.context.annotation.ConditionContext;
+import org.springframework.core.env.PropertyResolver;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+class OnKubernetesOrOpenshiftCondition extends SpringBootCondition {
```

Comment:
Again, I'd like to avoid having Openshift specific stuff in the core. They should go to a module of their own.

Having, said that I think that it might not be feasible to to really make sure if its vanilla. So we might need to check the behaviour to any working k8s environment will do. 

BTW, this is on me, cause I haven't thought of that detail, when we discussed about it. Sorry.

Reply:
No problem at all :)

I will move the Openshift stuff to a separate module as you mention.
And I understand correctly, since there is no way of guaranteeing that a k8s cluster is a vanilla cluster, I should remove the `requireVanilla` option from `ConditionalOnKubernetes` (along with the respective tests of course)?